### PR TITLE
pAI Journalism Update

### PR DESCRIPTION
**Pull request by RaveRadbury**
## About The Pull Request
Adding newscasters and cameras to pAI so they can make even more use of SS13's social features.

## Why It's Good For The Game
-Big Thought-
We're in a time of high population with many feeling unsatisfied with the available ghost roles. Adding more features to pAI's is a good way to ameliorate that. It's good for new and old players, especially the new ones as they can learn and participate in the culture of the game. Preserving the culture is important, right?

With decent ghost roles that are unlikely to cause incredible amounts of lag we might be able to argue for lifting the pop caps and having observers in. 

-Anyway-
This PR lets you make your pAI into your own personal publicist. Or combine these new abilities with the loudness booster and have a bard spreading word of your feats both through song and poem. These features don't get used enough and they are perfect for little gameboys that sit in PDA's. 

## Changelog
:cl: RaveRadbury, Art by Navi.OS
add: pAI Newscaster
add: pAI Camera
add: New pAI HUD
imageadd: unique pAI HUD theme
/:cl:

- [x] Add newscasters to pAI's
- [x] The newscaster is functional
- [x] Add cameras to pAI's
- [x] Update the pAI hud to include camera icons
- [x] Make sure camera works
- [x] Create pAI HUD in the first place
- [x] Remove old pAI Action HUD
- [x] Fix "Cannot interface with silicon photo uploading"
I'm choosing not to add the newscaster and camera to the software stack at this time.
 - I have a pAI HUD overhaul that I am partially integrating here and I need to look more into the possibility of modular huds as software is downloaded
 - Might as well let the new features be free for a second, it's not like they change much.
 - Were I to have the software integrated in the menus, the newscaster would just open the newcaster (behind the software menu, ew) anyway.
 - I want to make the camera software menu let you adjust the capture area but I haven't figured that out yet and pAI is dying for feautures.
 - I'd prefer to look at pricing in the scope of the entire stack of software and I'm already waiting on another software PR to go through.

### CODE REVIEW

**Comment by RaveRadbury** [comment]
I need to restore a bunch of stuff, so don't start snarking until it's all up, m'kay?

**Comment by Shdorsh** [comment]
Or we could just re-introduce circuits and make the circuit-bots mayb- **gets kicked out the window**

Just kidding. it's great to see a role that hasn't seen much update get some more features.
This makes me think, what about making arcade games downloadable for pAIs? Would keep the Ssethtide a bit more entertained at least, perhaps?

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -123,7 +123,7 @@
 			if ((M.client && M.machine == src))
 				is_in_use = TRUE
 				ui_interact(M)
-		if(isAI(usr) || iscyborg(usr) || IsAdminGhost(usr))
+		if(issilicon(usr) || IsAdminGhost(usr))
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/objs.dm]: Are you sure this is a safe change

Comment (RaveRadbury) [reply code/game/objects/objs.dm]: I made this change under the advisement of @AnturK

File: code/game/machinery/newscaster.dm [hunk 2]
```diff
@@ -791,6 +791,9 @@ GLOBAL_LIST_EMPTY(allCasters)
 		if(isAI(user))
 			var/mob/living/silicon/ai/R = user
 			targetcam = R.aicamera
+		else if(ispAI(user))
+			var/mob/living/silicon/ai/R = user
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/newscaster.dm]: this isn't typed correctly

Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/newscaster.dm]:
```suggestion
			var/mob/living/silicon/pai/R = user
```

File: code/_onclick/hud/pai.dm [hunk 3]
```diff
@@ -0,0 +1,134 @@
+/obj/screen/pai
+	icon = 'icons/mob/screen_pai.dmi'
+
+/obj/screen/pai/Click()
+	if(isobserver(usr) || usr.incapacitated())
+		return TRUE
```
Comment (ExcessiveUseOfCobblestone) [inline code/_onclick/hud/pai.dm]: what does return true do here?

Comment (ExcessiveUseOfCobblestone) [reply code/_onclick/hud/pai.dm]: nvm you use it for checks below

File: code/game/machinery/newscaster.dm (hunk 2 above)
Comment (RaveRadbury) [reply code/game/machinery/newscaster.dm]: Thanks for catching this!

**Comment by RaveRadbury** [comment]
Oh god why is it throwing errors from the wire fix PR.

**Comment by ExcessiveUseOfCobblestone** [comment]
whoops I think I used your version instead of master hehe

**Comment by RaveRadbury** [comment]
On thing that I'm not understanding is why git thinks that I replaced all of silicon/pai.dm? It's all the same stuff.

**Comment by ExcessiveUseOfCobblestone** [comment]
line endings

**Comment by RaveRadbury** [comment]
I'm gonna give this a fiddle, because I don't wanna get git blamed for this entire file.

**Comment by ExcessiveUseOfCobblestone** [comment]
wake me up inside (can't wake up)

**Comment by RaveRadbury** [comment]
It looks like when I did a pull to the pai-news-update branch it caused this. Should I just do a new PR or can I fix this somehow?

**Comment by ExcessiveUseOfCobblestone** [comment]
as long as the content is the same it's fine (it was when I checked it after you fixed it failing travis)

**Comment by RaveRadbury** [comment]
Content is all the same and trying to use `git revert` is driving me crazy. It should be good to go.

**Pull request merged**